frame/contracts/src/exec.rs

```diff
@@ -728,7 +728,9 @@ where
 	fn pop_frame(&mut self, persist: bool) {
 		// Revert the account counter in case of a failed instantiation.
 		if !persist && self.top_frame().entry_point == ExportedFunction::Constructor {
-			self.account_counter.as_mut().map(|c| *c = c.wrapping_sub(1));
+			if let Some(c) = self.account_counter.as_mut() {
+				*c = c.wrapping_sub(1);
+			}
```

Comment:
Why? It seems more simple before.

Reply:
Great question. This is actually from `option_map_unit_fn` which I have subsequently put on the allowed list because we do it a fair few times. 

"Checks for usage of option.map(f) where f is a function or closure that returns the unit type ()."

The rational is that an `if let` statement is more readable / clear. I personally agree on this one, but happy to ask the audience which they prefer: thumbs up, thumbs down - do we want this rule?

Reply:
> (arguably this is more of a stylistic rule)

All changes in this pr are only stylistic changes, nothing more.

Reply:
I prefer leaving it as is. But I can see why this can be a footgun. If I would have missed the `as_mut()` it would have been a bug. Clippy doesn't seem to to be able to differentiate whether the `Option` contains a reference or a moved value.

Reply:
I believe it was an oversight from the stdlib maintainers that they called this `map`. It more resembles a `for_each` in a sense that it does not need a `next`/`collect` to pull it, but it is always executed if the `Option` is a `Some`. The clippy rule just tries to avoid this surprise.

Reply:
This was no oversight. This is called `map` everywhere, because you `map` a value from x to y. `for_each` doesn't return anything.

Reply:
Okay, oversight is a bit too harsh indeed. When we map from any type to `()`, we are not using `map` to map a value to something, but to execute a closure on the value, like in `for_each`. And some people might get the wrong idea (especially in a `Vec<Option<Vec<Vec<T>>>>>` case that we often have without worrying about it too much) that map needs to be pulled on this `Option`, too.

Reply:
It looks like the consensus is that we don't want that rule. I will back out this change as its not relevant to the rule set chosen.